sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+
+    @staticmethod
+    def name():
+        return "fission"
+
+    @property
+    def config(self) -> FissionConfig:
+        return self._config
+
+    @staticmethod
+    def add_port_forwarding(port=5051):
+        podName = (
+            subprocess.run(
+                f"kubectl --namespace fission get pod -l svc=router -o name".split(),
+                stdout=subprocess.PIPE,
+            )
+            .stdout.decode("utf-8")
+            .rstrip()
+        )
+        subprocess.Popen(
+            f"kubectl --namespace fission port-forward {podName} {port}:8888".split(),
+            stderr=subprocess.DEVNULL,
+        )
+
+    def shutdown(self) -> None:
+        if self.config.shouldShutdown:
+            if hasattr(self, "httpTriggerName"):
+                subprocess.run(f"fission httptrigger delete --name {self.httpTriggerName}".split())
+            if hasattr(self, "functionName"):
+                subprocess.run(f"fission fn delete --name {self.functionName}".split())
+            if hasattr(self, "packageName"):
+                subprocess.run(f"fission package delete --name {self.packageName}".split())
+            if hasattr(self, "envName"):
+                subprocess.run(f"fission env delete --name {self.envName}".split())
+            stop_minikube()
+        self.storage.storage_container.kill()
+        logging.info("Minio stopped")
+
+    def get_storage(self, replace_existing: bool = False) -> PersistentStorage:
+        self.storage = Minio(self.docker_client)
+        return self.storage
+
+    def initialize(self, config: Dict[str, str] = None):
+        if config is None:
+            config = {}
+
+        check_if_minikube_installed()
+        check_if_k8s_installed()
+        check_if_helm_installed()
+        run_minikube()
+        Fission.add_port_forwarding()
+        sleep(5)
+
+    def package_code(self, benchmark: Benchmark) -> Tuple[str, int]:
+
+        benchmark.build()
+
+        CONFIG_FILES = {
+            "python": [
+                "handler.py",
+                "requirements.txt",
+                ".python_packages",
+                "build.sh",
+            ],
+            "nodejs": ["handler.js", "package.json", "node_modules"],
+        }
+        directory = benchmark.code_location
+        package_config = CONFIG_FILES[benchmark.language_name]
+        function_dir = os.path.join(directory, "function")
+        os.makedirs(function_dir)
+        minioConfig = open("./code/minioConfig.json", "w+")
+        minioConfigJson = {
+            "access_key": self.storage.access_key,
+            "secret_key": self.storage.secret_key,
+            "url": self.storage.url,
+        }
+        minioConfig.write(json.dumps(minioConfigJson))
+        minioConfig.close()
+        scriptPath = os.path.join(directory, "build.sh")
+        self.shouldCallBuilder = True
+        f = open(scriptPath, "w+")
+        f.write(
+            "#!/bin/sh\npip3 install -r ${SRC_PKG}/requirements.txt -t \
```

Comment:
This is not needed - our build image should handle that. OpenWhisk shows an example.

Reply:
yes, I create the zip package using the build image.,